python/lsst/daf/butler/core/_butlerUri/_butlerUri.py

```diff
@@ -653,6 +654,43 @@ def exists(self) -> bool:
         """
         raise NotImplementedError()
 
+    @classmethod
+    def mexists(cls, uris: Iterable[ButlerURI]) -> Dict[ButlerURI, bool]:
+        """Check for existence of multiple URIs at once.
+
+        Parameters
+        ----------
+        uris : iterable of `ButlerURI`
+            The URIs to test.
+
+        Returns
+        -------
+        existence : `dict` of [`ButlerURI`, `bool`]
```

Comment:
I think that with proper handling this (and the similar return for `Datastore.mexists()`) can be replaced by a simpler `set of ButlerURI`, but that can be left for later.

Reply:
I haven't yet understood how one set can do it. I can see how a found set and not-found set can do it.